pkg/logging/logging.go

```diff
@@ -133,6 +137,28 @@ func LoadLogConfig(dataStore, ns, id string) (LogConfig, error) {
 	return logConfig, nil
 }
 
+func loggingProcessAdapter(driver Driver, config *logging.Config) error {
+	var wg sync.WaitGroup
+	wg.Add(3)
+	stdout := make(chan string, 10000)
+	stderr := make(chan string, 10000)
+	processLogFunc := func(reader io.Reader, dataChan chan string) {
+		defer wg.Done()
+		scanner := bufio.NewScanner(reader)
+		for scanner.Scan() {
+			if scanner.Err() != nil {
+				return
+			}
+			dataChan <- scanner.Text()
+		}
+	}
+	go processLogFunc(config.Stdout, stdout)
+	go processLogFunc(config.Stderr, stderr)
+	go driver.Process(&wg, stdout, stderr)
```

Comment:
It's not a good idea to pass the outer WaitGroup instance into the inner function as delegating the control to Driver implementation.
Suggest:
```go
go func() {
	defer wg.Done()
	driver.Process(stdout, stderr)
} ()
```